src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useSignPermit2.tsx

```diff
@@ -0,0 +1,92 @@
+import { toAddressNList } from '@shapeshiftoss/chain-adapters'
+import type { TradeQuote, TradeQuoteStep } from '@shapeshiftoss/swapper'
+import assert from 'assert'
+import { useCallback, useMemo } from 'react'
+import { useErrorHandler } from 'hooks/useErrorToast/useErrorToast'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { assertGetEvmChainAdapter } from 'lib/utils/evm'
+import { selectPortfolioAccountMetadataByAccountId } from 'state/slices/selectors'
+import { selectHopSellAccountId } from 'state/slices/tradeQuoteSlice/selectors'
+import { tradeQuoteSlice } from 'state/slices/tradeQuoteSlice/tradeQuoteSlice'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+// handles allowance approval tx execution, fees, and state orchestration
+export const useSignPermit2 = (
+  tradeQuoteStep: TradeQuoteStep,
```

Comment:
Prefer we wait for a need to decouple this, thinking in yagni terms. Since this hook is reasonably at the "leaf" of the swapper context, this is a two-way door and can be broken out on demand if there's ever a need to 🙏

Reply:
yagni and on-demand sounds good to me, we may not need it indeed!